src/osd/osd_types.h

```diff
@@ -4117,7 +4117,7 @@ struct object_info_t {
   explicit object_info_t(bufferlist& bl) {
     decode(bl);
   }
-  object_info_t operator=(bufferlist& bl) {
+  object_info_t &operator=(bufferlist& bl) {
```

Comment:
@jimmyway please try compile the tree first. `PG::_scan_snaps()` uses this function, that's why i suggested use `decode()` directly.